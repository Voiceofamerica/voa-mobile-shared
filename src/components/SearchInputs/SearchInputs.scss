
@import '../../commonStyles/index.scss';

.inputsPill {
  border-radius: 0.8em;
  overflow: hidden;
  display: flex;
  background: $light-grey;

  .dropdownPill {
    position: relative;
    display: flex;
    width: 6em;
    border-radius: 0.8em;
    overflow: hidden;

    .dropdown {
      @include touchable();

      -webkit-appearance: none;
      -moz-appearance: none;
      font-size: 1em;
      border: none;
      padding: 0.3em;
      outline: none;
      background: $border-grey;
    }

    .dropdownArrow {
      display: flex;
      position: absolute;
      justify-content: flex-end;
      align-items: center;
      pointer-events: none;
      right: 0;
      top: 0;
      bottom: 0;
      font-size: 1.5em;
      width: 2em;
      background: linear-gradient(90deg, rgba($border-grey, 0), $border-grey 50%, $border-grey);

      [dir="rtl"] & {
        left: 0;
        right: auto;
        background: linear-gradient(-90deg, rgba($border-grey, 0), $border-grey 50%, $border-grey);
      }
    }
  }

  .searchInputContainer {
    position: relative;
    display: flex;
    flex: 1;
  }

  .searchInput {
    flex: 1;
    font-size: 1.2em;
    background: $light-grey;
    border: none;
    padding: 0 0.5em;
    outline: none;
  }

  .emptyText {
    pointer-events: none;
    position: absolute;
    display: flex;
    font-size: 1.2em;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    padding: 0 0.5em;
    color: rgba(0, 0, 0, 0.5);
    align-items: center;
  }
}