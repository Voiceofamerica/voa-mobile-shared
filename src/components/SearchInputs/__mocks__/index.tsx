
import * as React from 'react'

export default (props) => (
  React.createElement('search-input', { props }, props.children)
)
