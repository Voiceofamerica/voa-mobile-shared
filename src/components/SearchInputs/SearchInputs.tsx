
import * as React from 'react'

import { inputsPill, dropdownPill, dropdown, dropdownArrow, searchInputContainer, searchInput, emptyText } from './SearchInputs.scss'

export interface Props {
  zoneId: number
  query: string
  empty: string
  onZoneIdChange: (newZoneId: number) => void
  onQueryChange: (newQuery: string) => void
  categories: { id: string, name: string }[]
}

export default class SearchInputs extends React.Component<Props> {
  render () {
    const { zoneId, query, onZoneIdChange, onQueryChange, categories, empty } = this.props

    return (
      <div className={inputsPill}>
        <div className={dropdownPill}>
          <select value={zoneId} className={dropdown} onChange={ev => onZoneIdChange(parseInt(ev.currentTarget.value, 10))}>
            {
              categories.map(cat => (
                <option key={cat.id} value={cat.id}>{cat.name}</option>
              ))
            }
          </select>
          <span className={dropdownArrow}><i className='mdi mdi-chevron-down' /></span>
        </div>
        <div className={searchInputContainer}>
          <input autoFocus className={searchInput} value={query} onChange={ev => onQueryChange(ev.currentTarget.value)} />
          { query.length === 0 ? <div className={emptyText}>{empty}</div> : null }
        </div>
      </div>
    )
  }
}
