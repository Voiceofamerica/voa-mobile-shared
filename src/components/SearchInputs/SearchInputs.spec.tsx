
import * as React from 'react'
import { create } from 'react-test-renderer'
import { shallow } from 'enzyme'

import SearchInputs, { Props } from './SearchInputs'

const zones: Props['categories'] = [
  {
    id: '1',
    name: 'Zone 1',
  },
  {
    id: '2',
    name: 'Zone 2',
  },
  {
    id: '3',
    name: 'Zone 3',
  },
]

describe(`<${SearchInputs.name} />`, () => {
  describe('shapshots', () => {
    it('should render with empty text', () => {
      const element = create((
        <SearchInputs
          zoneId={1}
          query=''
          empty='Enter query'
          onZoneIdChange={jest.fn()}
          onQueryChange={jest.fn()}
          categories={zones}
        />
      ))

      expect(element).toMatchSnapshot()
    })

    it('should render without empty text but with the query', () => {
      const element = create((
        <SearchInputs
          zoneId={1}
          query='My Query'
          empty='Enter query'
          onZoneIdChange={jest.fn()}
          onQueryChange={jest.fn()}
          categories={zones}
        />
      ))

      expect(element).toMatchSnapshot()
    })

    it('should render with the selected category', () => {
      const element = create((
        <SearchInputs
          zoneId={3}
          query=''
          empty='Enter query'
          onZoneIdChange={jest.fn()}
          onQueryChange={jest.fn()}
          categories={zones}
        />
      ))

      expect(element).toMatchSnapshot()
    })
  })

  describe('onZoneIdChange', () => {
    const ids = [1, 2, 3]
    ids.forEach(expected => {
      it(`should be called whenever the category is changed to ${expected}`, () => {
        const onZoneIdChange = jest.fn()

        const item = shallow((
          <SearchInputs
            zoneId={3}
            query=''
            empty='Enter query'
            onZoneIdChange={onZoneIdChange}
            onQueryChange={jest.fn()}
            categories={zones}
          />
        ))

        item.find('select').simulate('change', {
          currentTarget: {
            value: `${expected}`,
          },
        })

        expect(onZoneIdChange).toHaveBeenCalledTimes(1)
        expect(onZoneIdChange).toHaveBeenCalledWith(expected)
      })
    })
  })

  describe('onQueryChange', () => {
    const queries = ['query 1', 'QUERY 2', 'Query 3']
    queries.forEach(expected => {
      it(`should be called whenever the query is changed to ${expected}`, () => {
        const onQueryChange = jest.fn()

        const item = shallow((
          <SearchInputs
            zoneId={3}
            query=''
            empty='Enter query'
            onZoneIdChange={jest.fn()}
            onQueryChange={onQueryChange}
            categories={zones}
          />
        ))

        item.find('input').simulate('change', {
          currentTarget: {
            value: expected,
          },
        })

        expect(onQueryChange).toHaveBeenCalledTimes(1)
        expect(onQueryChange).toHaveBeenCalledWith(expected)
      })
    })
  })
})
