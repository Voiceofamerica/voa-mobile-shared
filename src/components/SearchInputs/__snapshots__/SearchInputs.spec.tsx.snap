// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInputs /> shapshots should render with empty text 1`] = `
<div
  className="inputsPill"
>
  <div
    className="dropdownPill"
  >
    <select
      className="dropdown"
      onChange={[Function]}
      value={1}
    >
      <option
        value="1"
      >
        Zone 1
      </option>
      <option
        value="2"
      >
        Zone 2
      </option>
      <option
        value="3"
      >
        Zone 3
      </option>
    </select>
    <span
      className="dropdownArrow"
    >
      <i
        className="mdi mdi-chevron-down"
      />
    </span>
  </div>
  <div
    className="searchInputContainer"
  >
    <input
      autoFocus={true}
      className="searchInput"
      onChange={[Function]}
      value=""
    />
    <div
      className="emptyText"
    >
      Enter query
    </div>
  </div>
</div>
`;

exports[`<SearchInputs /> shapshots should render with the selected category 1`] = `
<div
  className="inputsPill"
>
  <div
    className="dropdownPill"
  >
    <select
      className="dropdown"
      onChange={[Function]}
      value={3}
    >
      <option
        value="1"
      >
        Zone 1
      </option>
      <option
        value="2"
      >
        Zone 2
      </option>
      <option
        value="3"
      >
        Zone 3
      </option>
    </select>
    <span
      className="dropdownArrow"
    >
      <i
        className="mdi mdi-chevron-down"
      />
    </span>
  </div>
  <div
    className="searchInputContainer"
  >
    <input
      autoFocus={true}
      className="searchInput"
      onChange={[Function]}
      value=""
    />
    <div
      className="emptyText"
    >
      Enter query
    </div>
  </div>
</div>
`;

exports[`<SearchInputs /> shapshots should render without empty text but with the query 1`] = `
<div
  className="inputsPill"
>
  <div
    className="dropdownPill"
  >
    <select
      className="dropdown"
      onChange={[Function]}
      value={1}
    >
      <option
        value="1"
      >
        Zone 1
      </option>
      <option
        value="2"
      >
        Zone 2
      </option>
      <option
        value="3"
      >
        Zone 3
      </option>
    </select>
    <span
      className="dropdownArrow"
    >
      <i
        className="mdi mdi-chevron-down"
      />
    </span>
  </div>
  <div
    className="searchInputContainer"
  >
    <input
      autoFocus={true}
      className="searchInput"
      onChange={[Function]}
      value="My Query"
    />
  </div>
</div>
`;
